org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/AdempiereIdGenerator.java

```diff
@@ -106,6 +107,8 @@ private static String getWidgetName(String widgetClass) {
 	}
 
 	public static String escapeId(String prefix) {
+		if (! "true".equals(System.getProperty("UseAdempiereIdGenerator.escapeId")))
+			return UUID.randomUUID().toString();
 		Pattern pattern = Pattern.compile("[^a-zA-Z_0-9]");
 		Matcher matcher = pattern.matcher(prefix);
```

Comment:
I think it is better to skip those setid call if it is not running selenium, i.e `if ("true".equals(System.getProperty("ZkUnitTest") setId(AdempiereIdGenerator.escapeId(...))`. The proposed patch work but it is not good for performance.

Reply:
Hi @hengsin - I tested first simply returning null when not selenium, and it worked fine for the case reported on this ticket, and there are more setId that seems like can be skipped safely.

But I saw a lot of code also using the AdempiereIdGenerator.escapeId for setting setWidgetAttribute("instanceName"...

And I'm not sure how that setWidgetAttribute can affect it if we skip it completely too.

Can those be skipped safely too?

Also, I think it would be good if we collect in a single class all the System.getProperty spread in the system, but not sure how must be the name to start such class and where it must  be located.

Reply:
Those setWidgetAttribute call is for selenium test too. This is even safer to skip than the setId call.

Reply:
Ready @hengsin - changed as per your suggestion